tools/python/stylesheet/color_matrix_builder.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python2.7
+
+from __future__ import print_function
+
+import drules_struct_pb2 as proto
+from image_maker import argb_to_tuple, ImageMaker
+
+import re
+import sys
+from collections import defaultdict
+from os import path
+
+
+ALLOWED_FIELDS_RE = re.compile("^[a-z]")
+TYPE_RE = re.compile(".*Proto$")
+DROP_NUMBER = re.compile("_\d*$")
+
+
+def is_iterable(obj):
+    if isinstance(obj, unicode):
+        return False
+    try:
+        some_object_iterator = iter(obj)
+        return True
+    except TypeError:
+        return False
+
+
+class MatrixBuilder:
+    def __init__(self):
+        self.transformation_table = None  # {(color1, color2): [Element]}
+        self.inverted_transformation_table = {}  # {element_key: Number_in_transformation_table}
+        self.paths = []
+        self.drules = []
+        self.imgs = []
+        self.all_elements = defaultdict(dict)  # {path_index: {[(el.name, el.priority), (...)] : colour}}
+
+
+    def invert_transformation_table(self):
+        for element, i in zip(self.transformation_table.iteritems(), range(0, len(self.transformation_table))):
+            for key in element[1]:
+                self.inverted_transformation_table[key] = i
```

Comment:
if there is a possibility of a large seq of keys you may consider:

keys_with_indices = izip(element[1], repeat(i))
self.inverted_transformation_table.update(keys_with_indices)

for a better performance